airbyte-connector-builder-server/connector_builder/impl/default_api.py

```diff
@@ -108,12 +111,14 @@ async def read_stream(self, stream_read_request_body: StreamReadRequestBody = Bo
         :return: Airbyte record messages produced by the sync grouped by slice and page
         """
         adapter = self._create_low_code_adapter(manifest=stream_read_request_body.manifest)
+        schema_inferrer = SchemaInferrer()
```

Comment:
this class is stateful right? we'll need to reset it or create a new SchemaInferrer for every read to avoid leaking the type of records produced from a different configuration

Reply:
isn't this a local variable in the scope of the current `read_stream` invocation? If that's the case there should be a new schema inferrer for every call of `read_stream` which is exactly what we want, right?